pkg/k8s/apis/cilium.io/client/crds/v2/ciliumlocalredirectpolicies.yaml

```diff
@@ -262,7 +265,11 @@ spec:
                     - serviceName
                     type: object
                 type: object
+                x-kubernetes-validations:
+                - message: redirectFrontend is immutable
+                  rule: self == oldSelf
               skipRedirectFromBackend:
+                default: false
```

Comment:
I'm curious: why is this explicit `default: false` required?

Reply:
This is because if `default: false` is not explicitly set, there's no way to prevent the modification of `skipRedirectFromBackend` from `""` to `true`.

Example:

``` shell
#  kubectl apply -f https://raw.githubusercontent.com/cilium/cilium/1.15.6/examples/kubernetes-local-redirect/lrp-addrmatcher.yaml
ciliumlocalredirectpolicy.cilium.io/lrp-addr create
```

``` diff
diff --git a/examples/kubernetes-local-redirect/lrp-addrmatcher.yaml b/examples/kubernetes-local-redirect/lrp-addrmatcher.yaml
index 547a32a4a4..b1539d802e 100644
--- a/examples/kubernetes-local-redirect/lrp-addrmatcher.yaml
+++ b/examples/kubernetes-local-redirect/lrp-addrmatcher.yaml
@@ -3,6 +3,7 @@ kind: CiliumLocalRedirectPolicy
 metadata:
   name: "lrp-addr"
 spec:
+  skipRedirectFromBackend: true
   redirectFrontend:
     addressMatcher:
       ip: "169.254.169.254"
```

``` shell
# kubectl apply -f examples/kubernetes-local-redirect/lrp-addrmatcher.yaml
ciliumlocalredirectpolicy.cilium.io/lrp-addr configured



# kubectl get clrp lrp-addr -oyaml
apiVersion: cilium.io/v2
kind: CiliumLocalRedirectPolicy
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"cilium.io/v2","kind":"CiliumLocalRedirectPolicy","metadata":{"annotations":{},"name":"lrp-addr","namespace":"default"},"spec":{"redirectBackend":{"localEndpointSelector":{"matchLabels":{"app":"proxy"}},"toPorts":[{"port":"80","protocol":"TCP"}]},"redirectFrontend":{"addressMatcher":{"ip":"169.254.169.254","toPorts":[{"port":"8080","protocol":"TCP"}]}},"skipRedirectFromBackend":true}}
  creationTimestamp: "2024-07-11T02:16:38Z"
  generation: 2
  name: lrp-addr
  namespace: default
  resourceVersion: "1349863"
  uid: 4f1f632b-31c3-414c-9b28-0f239a377d73
spec:
  redirectBackend:
    localEndpointSelector:
      matchLabels:
        app: proxy
    toPorts:
    - port: "80"
      protocol: TCP
  redirectFrontend:
    addressMatcher:
      ip: 169.254.169.254
      toPorts:
      - port: "8080"
        protocol: TCP
  skipRedirectFromBackend: true
``` shell